Add more XML libraries to comparison

Mingun opened the pull request:
We still 2-3x faster than `xmlparser`, `xml5ever` on the third, `xml-rs` is outsider.

Rough run on my notebook
```
low-level API/fast_xml  time:   [248.44 us 261.39 us 276.66 us]
                        change: [+75.497% +104.05% +139.32%] (p = 0.00 < 0.05)
                        Performance has regressed.
Found 10 outliers among 100 measurements (10.00%)
  3 (3.00%) high mild
  7 (7.00%) high severe
low-level API/xml_rs    time:   [14.415 ms 14.940 ms 15.524 ms]
                        change: [+28.625% +34.300% +39.581%] (p = 0.00 < 0.05)
                        Performance has regressed.
Found 10 outliers among 100 measurements (10.00%)
  8 (8.00%) high mild
  2 (2.00%) high severe
low-level API/xml5ever  time:   [5.3598 ms 5.4856 ms 5.6368 ms]
                        change: [-13.234% -9.9922% -6.4032%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 5 outliers among 100 measurements (5.00%)
  4 (4.00%) high mild
  1 (1.00%) high severe
low-level API/xmlparser time:   [682.64 us 688.38 us 695.22 us]
Found 10 outliers among 100 measurements (10.00%)
  6 (6.00%) high mild
  4 (4.00%) high severe
```

@jdm, you are merging last PRs in xml5ever -- if you or other of your team interested, an you look at the benchmark code? I would like to count tags as the first two benchmarks do.

@RazrFalcon, if you interested, could you look at benchmark code for xmlparser? I'm not sure that it is idiomatic code, moreover, I would like to count tags as the first two benchmarks do.

Mingun committed:
Add xml5ever to comparison

It is number 17 (at time of commit) in "Parser implementations" topic
with 137 usages (5 direct) according to https://lib.rs/crates/xml5ever

Mingun committed:
Add xmlparser to comparison

It is number 38 (at time of commit) in "Parser implementations" topic
with 620 usages (15 direct) according to https://lib.rs/crates/xmlparser

RazrFalcon commented:
Hi! I would suggest benchmarking agains roxmltree. xmlparser is very low-level.

dralley commented:
FYI benchmarks tend to be profoundly unreliable on laptops, I've seen huge swings just from power / thermal management. (Not that it matters as much when comparing different libraries, these gaps are pretty large)

Mingun committed:
Add xml_oxide to comparison

It is number 143 (at time of commit) in "Parser implementations" topic
without usages according to https://lib.rs/crates/xml_oxide

Mingun committed:
Add rapid-xml to comparison

It is number 320 (at time of commit) in "Parser implementations" topic
without usages according to https://lib.rs/crates/rapid-xml

Mingun committed:
Add maybe_xml to comparison

It is number 264 (at time of commit) in "Parser implementations" topic
without usages according to https://lib.rs/crates/maybe_xml

Mingun committed:
Add RustyXML to comparison

Usages: 63 (16 direct) according to https://lib.rs/crates/RustyXML

Mingun commented:
Merged in https://github.com/tafia/quick-xml/pull/386 as quick-xml resurrected
